### PR TITLE
Updates for Micrometer 1.0.0-rc.2

### PR DESCRIPTION
**Pull request by jkschneider**
Fixes and improvements made for rc.2 are listed [here](https://github.com/micrometer-metrics/micrometer/milestone/13?closed=1). Significantly, the gap on cache metrics has been closed.

The impact on Boot 2 is:

1. A new autoconfig for the StatsD registry.
2. A new protocol property for Graphite.
3. A reworked metrics actuator endpoint.

### CODE REVIEW

File: spring-boot-project/spring-boot-dependencies/pom.xml [hunk 1]
```diff
@@ -113,7 +113,7 @@
 		<logback.version>1.2.3</logback.version>
 		<lombok.version>1.16.18</lombok.version>
 		<mariadb.version>2.1.2</mariadb.version>
-		<micrometer.version>1.0.0-rc.1</micrometer.version>
+		<micrometer.version>1.0.0-SNAPSHOT</micrometer.version>
```
Comment (jkschneider) [inline spring-boot-project/spring-boot-dependencies/pom.xml]: If you are satisfied with what's been done in the rest of this PR, I can go ahead and release 1.0.0-rc.2

**Comment by jkschneider** [comment]
cc/ @philwebb @wilkinsona

**Comment by jkschneider** [comment]
Sample of the updated metrics endpoint:

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/statsd/StatsdExportConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+	@Bean
+	@ConditionalOnProperty(value = "spring.metrics.statsd.enabled", matchIfMissing = true)
+	public MetricsExporter statsdExporter(StatsdConfig statsdConfig, Clock clock) {
+		return () -> new StatsdMeterRegistry(statsdConfig, clock);
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public Clock clock() {
```
Comment (shakuzen) [inline spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/statsd/StatsdExportConfiguration.java]: See #10436

Comment (wilkinsona) [reply spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/statsd/StatsdExportConfiguration.java]: 👍 I'll take care of that one as part of merging this

**Comment by wilkinsona** [comment]
This looks good to me.

I've split it up into separate commits for each change here: https://github.com/wilkinsona/spring-boot/tree/gh-10519. The change the enables the metrics endpoint by default isn't there at the moment as I think it warrants separate consideration. I've opened #10525.

The only piece I'm not sure about is the removal of the requested tags from the available tags in the metrics endpoint:

```java
tags.forEach(t -> availableTags.remove(t.getKey()));
```

It feels a little odd to me to successfully use a particular tag as a query and then to not see that tag listed in the available tags for the metric. I wonder if this line should be removed and the tests updated accordingly?

**Comment by wilkinsona** [comment]
Aah, I see. Thanks for the explanation. That makes sense. Let's leave it as it is.

**Pull request closed without merging**